src/object.d

```diff
@@ -3190,11 +3199,24 @@ void __ctfeWriteln(T...)(auto ref T values) { __ctfeWrite(values, "\n"); }
  * Create RTInfo for type T
  */
 
+template RTInfoImpl(size_t[] pointers)
+{
+    immutable size_t[pointers.length] data = pointers[];
+    immutable RTInfoImpl = data.ptr;
```

Comment:
>> Arrays (e.g. of strings) can often get quite large. But I'm ok with leaving this optimization for later.
> 
> But that's an array of a simple type, not a large object. You can squeeze the pointer bitmap for 16-byte into 2 bits. The threshold for replacing a pointer bitmap with an 8-byte pointer would be a bitmap with 64 bits, i.e. an object with 512 bytes, those are rare.

I'm talking about the pointer bitmap `is_pointer` that lives next to the memory pool and indicates whether a memory word is a pointer or not, word by word (not the data stored in RTInfo). For an array the RTInfo pointer bitmap of the element type is repeated again and again in the `is_pointer` memory. If we know more about the array, and we restrict the functionality of `gc_emplace`, we might be able to avoid that.